.github/workflows/ci.yml

```
@@ -80,6 +80,7 @@ jobs:
           npm run test:ci
 
   automerge:
+    if: ${{ github.event_name == 'pull_request' || github.event.inputs.pr-number != '' }}
```

```suggestion
    if: github.actor == 'dependabot[bot]'
```
Dependabot PRs is the only thing that is automerged by this action

---

what about the workflow dispatch? If I do it like your suggestion, then you can not manually call it under the actions tab, because your personal account would be the actor, correct?

---

Workflow_dispatch isn't in the list of events that will trigger this workflow so this isn't an issue.

---

`github.actor` is not guarantee to be `dependabot[bot]`. 
Anyone who re-run the failed workflow will change the `actor` to that person.

---

If you rerun the Workflow, do you expect that its automatically merged? Or would you expect to have still full control and press after successfull run on the merge button?

---

Yes, I expect it will auto merged.
Sometimes, there are flaky test that require manual re-run. In that case, this change just block the auto merge.

---

@climba03003 not 100% on this but just about to head out so will dig further tomorrow

---

@Fdawgs 
For example, this two `dependabot` PR action actor is me.
I re-run it because something is failed.
https://github.com/fastify/fastify/actions/runs/1256679962
https://github.com/fastify/fastify/actions/runs/1256679962

---

Thanks @climba03003, dug through the docs as well and you're right.